metacat-common-server/src/main/java/com/netflix/metacat/common/server/connectors/SpringConnectorFactory.java

```diff
@@ -30,20 +30,24 @@
  */
 
 public abstract class SpringConnectorFactory implements ConnectorFactory {
-    protected final String catalogName;
     protected final AnnotationConfigApplicationContext ctx;
+    private final String catalogName;
```

Comment:
It seems to me rather than passing `catalogName` and `catalogShardName` to every class through the whole system (with the potential to want to pass more information later) it would be better to abstract these behind a class which encapsulates them so you only have to pass one object around and it can easily be extended without touching as many classes next time